plugins/embark-assistant/matcher.cpp

```diff
@@ -475,7 +475,8 @@ namespace embark_assist {
                 return;  //  We're at the world edge, so no incursions from the outside.
             }
 
-            if (!&survey_results->at(fetch_x).at(fetch_y).surveyed) {
+            if (!survey_results->at(fetch_x).at(fetch_y).surveyed) {
```

Comment:
Was this testing the memory address of the surveyed boolean? What was it supposed to catch?

Reply:
yea, don't know. I've presumed the `&` was a typo

Reply:
@PatrikLundell would you mind confirming that the new logic is correct?

Reply:
It wasn't a typo, but rather poor knowledge of High Level Assembly, a.k.a. C(++). The intention was to check the field, not any address.

Checking the address would never fail, as things would blow up with a segfault before it reached it, as far as I can see.

Reply:
> The intention was to check the field

Ok. Same outcome as if it had been a typo, off by one extra character.

Reply:
Oh, one thing I'm still curious about though.. why haven't we seen a problem with whatever error this was intended to prevent?

Reply:
I should clarify - my question is whether the newly-reachable branch (setting `*failed_match = true;`) _should_ be reachable if `.surveyed` is false, or if `.surveyed` is true or something else.

Reply:
Yes. If no survey has been performed the data needed for incursion processing hasn't been collected, and so the code should return with a failed match. If the data has been collected, incursion processing should be performed to evaluate whether a match actually is present.
Performing the incursion processing without (a complete set of) incursion data will result in some false positives and some false negatives, partially depending on the query (it's probably possible to make a query that gets both kinds of false results).